src/server/v2/resources/jobs.py

```diff
@@ -626,6 +636,13 @@ def post(self):
         new_job.arch = artifact_record.arch
         current_app.logger.info(f"architecture: {new_job.arch}")
 
+        # change the file name to match the architecture of the image and recipe, if custom ignore and pass
+        default_file_name = ARCH_TO_KERNEL_FILE_NAME.get(new_job.arch)
+        if (default_file_name
+            and new_job.kernel_file_name in ARCH_TO_KERNEL_FILE_NAME.values()
+            and new_job.kernel_file_name != default_file_name):
+            new_job.kernel_file_name = default_file_name
```

Comment:
I don't think this is quite right...

You are saying if they have passed something in, but it isn't the default, it will get overridden with the default value. This way the user can never set something other than what we call the default.

What we want, is if they enter something explicitly we should take that. If they don't enter anything, we should default based on the arch.